build.sbt

```
@@ -309,19 +313,50 @@ lazy val core = (project in file("core"))
 lazy val root = RootProject(file("."))
 
 lazy val chisel = (project in file("."))
-  .enablePlugins(ScalaUnidocPlugin)
   .settings(commonSettings: _*)
   .settings(chiselSettings: _*)
-  .settings(publishSettings: _*)
+  .settings(
+    // Published as part of unipublish
+    publish / skip := true
+  )
   .settings(usePluginSettings: _*)
   .dependsOn(macros)
   .dependsOn(core)
   .dependsOn(firrtl)
   .aggregate(macros, core, plugin, firrtl)
   .settings(warningSuppression: _*)
   .settings(fatalWarningsSettings: _*)
+
+
+def addUnipublishDeps(proj: Project)(deps: Project*): Project =
+  deps.foldLeft(proj) { case (p, dep) =>
+    //p.dependsOn(dep % "compile-internal")
+    p
+     .settings(
+       libraryDependencies ++= (dep / libraryDependencies).value,
+       Compile / packageBin / mappings ++= (dep / Compile / packageBin / mappings).value,
+       Compile / packageSrc / mappings ++= (dep / Compile / packageSrc / mappings).value,
+     )
+  }
+
+// This is a pseudo-project that unifies all compilation units (excluding the plugin) into a single artifact
+// It should be used for all publishing and MiMa binary compatibility checking
+lazy val unipublish =
+  addUnipublishDeps(project in file("unipublish"))(
+    firrtl, macros, core, chisel
+  )
+  .aggregate(plugin) // Also publish the plugin when publishing this project
+  .settings(name := (chisel / name).value)
+  .enablePlugins(ScalaUnidocPlugin)
+  .settings(commonSettings: _*)
+  .settings(publishSettings: _*)
+  .settings(usePluginSettings: _*)
+  .settings(warningSuppression: _*)
+  .settings(fatalWarningsSettings: _*)
   .settings(
-    mimaPreviousArtifacts := Set(),
+    mimaPreviousArtifacts := Set("org.chipsalliance" %% "chisel" % "5.0-SNAPSHOT"),
```

```suggestion
    mimaPreviousArtifacts := Set(),
```
accidentally left it from testing (this does work with MiMa).